tests/testsuite.js

```diff
@@ -0,0 +1,90 @@
+var startActiveXObjectSetTests = (function () {
+    var testLogger;
+
+    if (typeof console === 'object') {
+        testLogger = console;
+    } else if (typeof WScript !== 'undefined') {
+        testLogger = {
+            error: function () {
+                var args = Array.prototype.slice.call(arguments);
+                WScript.StdErr.WriteLine('Error: ' + args.join(' '));
+            },
+            log: function () {
+                var args = Array.prototype.slice.call(arguments);
+                WScript.Echo('Info: ' + args.join(' '));
+            }
+        };
+    } else {
+        throw new Error('Unknown environment');
+    }
+
+    function assertLog(exp, message) {
+        if (exp === true) {
+            testLogger.log(message);
+            return;
+        }
+        testLogger.error(message);
+    }
+
+    function getWSCObject(wscPath) {
+        var fso = new ActiveXObject('Scripting.FileSystemObject');
+        wscPath = fso.GetAbsolutePathName(wscPath);
+        return GetObject('Script:' + wscPath);
```

Comment:
Just out of curiosity, where is the `Script:` syntax documented?

Reply:
In Microsoft terms `Script` is a "Moniker" to a Windows Script Component. Since Microsoft destroyed almost all WSH related documents, it's hard to find online. Fortunately there's an [article](https://docs.microsoft.com/en-us/previous-versions/tn-archive/ee692823(v=technet.10)#calling-a-script-component-in-a-script) left that you can refer to.